tb_plugin/torch_tb_profiler/profiler/gpu_metrics_parser.py

```diff
@@ -0,0 +1,175 @@
+from .range_lib import *
+from .trace import EventTypes
+from .. import utils
+
+logger = utils.get_logger()
+
+
+# For calculating GPU utilization, and approximated SM efficiency.
+class GPUMetricsParser(object):
+    def __init__(self):
+        # All gpu ids that used by any kernel.
+        self.gpu_ids = set()
+        # For calculating GPU utilization.
+        self.kernel_ranges_per_device = []
+        self.gpu_utilization = []
+        self.gpu_util_timeline_unit_size = 0
+        self.gpu_util_timeline_unit_name = ""
+        self.gpu_util_json = None
+        # For calculating approximated SM efficiency.
+        self.blocks_per_sm_per_device = []
+        self.avg_approximated_sm_efficency_per_device = []
+        self.approximated_sm_efficency_ranges = []
+        self.gpu_sm_efficiency_json = None
+        # For calculating averaged occupancy.
+        self.occupancy_per_device = []
+        self.avg_occupancy_per_device = []
+
+    def calculate_gpu_utilization(self, steps_start_time, steps_end_time):
+        # Make bucket_size to 10-power's of us, and number of buckets to (10, 100].
+        # 10-power's of us, in order to straight forward for user to understand.
+        # If number of buckets are too many, the value of gpu utilization will be either 0 or 1.
+        def get_bucket_info(range_micro_seconds):
+            max_buckets = 100
+            bucket_size = 1
+            while range_micro_seconds / bucket_size > max_buckets:
+                bucket_size *= 10
+            buckets = int(range_micro_seconds / bucket_size)
+            unit = bucket_size
+            unit_str = "us"
+            if unit >= 1000:
+                unit /= 1000
+                unit_str = "ms"
+                if unit >= 1000:
+                    unit /= 1000
+                    unit_str = "s"
+            return int(bucket_size), int(buckets), int(unit), unit_str
+
+        def build_trace_counter(gpu_id, start_time, counter_value):
```

Comment:
need the logic be moved to plugin.py to speed up the loading time?

Reply:
Do you mean lazy initialization, delay it to just on calling trace view?

Reply:
just curious the reason that one part json generation is in plugin.py and another part is in gpu_metrics_parser.py

Reply:
Done. Moved to plugin.py, now consistent.